keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/MetropolisHastings.java

```diff
@@ -62,113 +57,51 @@ public static MetropolisHastings withDefaultConfig(KeanuRandom random) {
     @Getter
     @Setter
     @Builder.Default
-    private boolean useCacheOnRejection = DEFAULT_USE_CACHE_ON_REJECTION;
+    private ProposalRejectionStrategy rejectionStrategy = DEFAULT_REJECTION_STRATEGY;
+
+    @Getter
+    @Setter
+    @Builder.Default
+    private LogProbCalculationStrategy logProbCalculationStrategy = DEFAULT_LOG_PROB_CALCULATION_STRATEGY;
+
+    @Getter
+    @Setter
```

Comment:
I think we should remove `@Setter` from all of the variables in this class. The object can be immutable.

Reply:
done